src/Microsoft.AspNetCore.SystemWebAdapters/Adapters/RemoteAppServerExtensions.Framework.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.SystemWebAdapters;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace System.Web;
+
+public static class RemoteAppServerExtensions
+{
+    /// <summary>
+    /// Add configuration for connecting to a remote app for System.Web extensions that require a remote
+    /// ASP.NET app such as remote app authentication or remote app session sharing.
+    /// </summary>
+    public static ISystemWebAdapterBuilder AddRemoteAppServer(this ISystemWebAdapterBuilder builder, Action<ISystemWebAdapterRemoteServerAppBuilder> configure)
```

Comment:
Having both a builder callback and a builder return value on the same API is an odd mixture of patterns. Though I can also appreciate that returning ISystemWebAdapterRemoteServerAppBuilder here could have also been confusing and would break the ISystemWebAdapterBuilder chaining.

Reply:
yeah - I had the same thought, but seemed to make more sense than having a new builder. We can discuss it at the API review